Integrations/src/main/java/io/deephaven/integrations/learn/Scatterer.java

```diff
@@ -0,0 +1,54 @@
+package io.deephaven.integrations.learn;
+
+import io.deephaven.base.verify.Require;
+import io.deephaven.datastructures.util.CollectionUtil;
+import org.jpy.PyObject;
+
+import java.util.ArrayList;
+
+/**
+ * Scatterer applies scatter functions to the result of a deferred calculation and feeds the scattered data back into a table.
+ */
+public class Scatterer {
+
+    private final Output[] outputs;
+
+    /**
+     * Creates a new Scatterer.
+     *
+     * @param outputs array of Outputs that determine how data will be scattered back to the table.
+     */
+    public Scatterer(Output[] outputs) {
+
+        Require.neqNull(outputs, "outputs");
+
+        this.outputs = outputs;
+    }
+
+    /**
+     * Applies the scatter function of a particular output to the result of the deferred calculation.
+     *
+     * @param idx   index of the output to scatter back into the table.
+     * @param fo    FutureOffset that contains the results of the deferred calculation as well as the index to access
+     *              that calculation.
+     * @return subset of result that can be put back into the table.
+     */
+    public PyObject scatter(int idx, FutureOffset fo) {
+        return outputs[idx].getScatterCaller().apply(fo.getFutureGet(), fo.getOffset());
+    }
+
+    /**
+     * Generates a query string for each of these Outputs.
+     *
+     * @return list of query strings to be used in .update() call
+     */
+    public String[] generateQueryStrings() {
+        ArrayList<String> queryStrings = new ArrayList<>();
+
+        for (int i = 0; i < outputs.length; i++) {
+            queryStrings.add(String.format("%s = scatterer.scatter(%d, FutureOffset)", outputs[i].getColName(), i));
```

Comment:
1) Consider passing in the future offset column name.  As things are, the column names are spread apart in the code, so someone may not recoginzie all of the places where changes need to be made, if the code is modified in the future.
2) Consider renaming it to `__FutureOffset` to avoid possible collisions with user column names.

Reply:
??

Reply:
"FutureOffset" as an argument of `generateQueryStrings`?

Reply:
1) Here you have created a query string `"%s = scatterer.scatter(%d, FutureOffset)"`.  In this query string `FutureOffset` is a column name.  If someone changes the part of the code where the column is created, the code will break because this part of the code was not changed.  You are better off doing `"%s = scatterer.scatter(%d, %d)"` and passing the column name into this function.

2) There is a non-zero chance that the user already has a column named `FutureOffset`.  This should be worked around.  Either use a less popular name like `__FutureOffset`, or check to see if `__FutureOffset` is in use.  If it is in use, then try `__FutureOffset<number>` until you find a name without a conflict.  Such a strategy requires that you can handle different column names in this function.